src/components/interactive/interactive.component.tsx

```diff
@@ -5,21 +5,21 @@ import { useBoundingClientRect } from "@/hooks/use-bounding-client-rect";
 import { clamp } from "@/utils/clamp";
 
 interface IInteractiveProps {
-  readonly onCoordinateChange: (x: number, y: number) => void;
+  readonly onCoordinateChange: (x: number, y: number, final?: boolean) => void;
```

Comment:
Let's switch the order of the arguments and put `final` first. This will help us get rid of the ugly `_`-prefix style for unused arguments in the components using this event.
Also, `final` shouldn't be optional since its default value is `false`, meaning it will always be of `boolean` type.

Reply:
gotcha! yeah, in the end it is a tradeoff: `final` at the end requires the `_y` arg is not used and `final` at the beginning can't make it optional

Reply:
wondering if a object param instead might be a better option than the above?